great_expectations/datasource/fluent/fluent_base_model.py

```diff
@@ -294,3 +301,28 @@ def _recursively_set_config_value(
                 data[i] = v.get_config_value(config_provider)
             elif isinstance(v, (MutableMapping, MutableSequence)):
                 return _recursively_set_config_value(v, config_provider)
+
+
+def _update__fields_set__on_truthyness(model: FluentBaseModel, field_name: str) -> None:
+    """
```

Comment:
@Kilo59 I am approving, because I do not have a better proposal at the moment, other than to express that being overly invested in the inner workings of a framework (`Pydantic` in this case) may cause maintenance difficulties in the future (e.g., if the framework's behavior changes).  In addition, the use of `getattr()` masks the specifics of the the field being used, and while appropriate in certain situation may cause unforeseen results, if used on unintended field names (e.g., by accident) -- @cdkini may wish to offer an opinion on `getattr()`.  Finally, a question: is the absence of `assets` a normal situation, or a corner case?  If the former, then perhaps the `debug` log messages may not be needed.  Thank you.

Reply:
>Finally, a question: is the absence of assets a normal situation, or a corner case?
I would say it's very normal. A Datasource can be created without any `assets`.

Reply:
Hence, the log messages are not needed perhaps?

Reply:
I would prefer to keep the debug logs.
We log at `warning` and above, users won't see this.